### ISSUE TITLE
Clarification Needed on Impact of Regularization on Loss Surface

### ISSUE THREAD

**Issue by spontanurlaub** [udlbook/udlbook#90]
I'd like to request some clarification on a statement made on page 403:

> Another possible explanation for the ease with which models are trained is that regularization makes the loss surface flatter and more convex.

While I understand that L2 regularization (or weight decay) adds a convex term λ∣∣w∣∣^2 to the loss function, thereby potentially smoothing out the loss surface, it also modifies the Hessian matrix with the addition of the term 2λI. This indeed lends the function a more convex nature.

However, my confusion arises when considering other regularization methods, such as Dropout and Early Stopping. My current understanding is that these techniques primarily influence the trajectory of the optimizer on the loss surface, preventing it from settling into deep local minima. However, I'm not convinced that they inherently modify the shape or convexity of the loss surface itself.

Could you please elaborate on this, or point to further resources that shed light on this topic?

**Comment by udlbook** [comment]
Hey, apologies for the really long delay.  I've been taking some time off away from the book after working on it for a couple of years.  Back now.  I agree with your point and I'll change statement to be about L2 (weight decay) regularization.  If you send your full name to the address on the front of the pdf, I'll add you to the acknowledgements.

Dropout can be thought of as injecting noise and it's possible that this means that the expected gradient is smoother, but I don't know of a proof of this.  I don't think that early stopping can be thought of in terms of smoothing the gradient.
